lib/controller/groepen/AbstractGroepenController.php

```diff
@@ -484,151 +430,150 @@ public function converteren(array $selection) {
+		$groep = $this->model->retrieveByUUID($id);
+		if ($groep and property_exists($groep, 'aanmelden_tot') and time() <= strtotime($groep->aanmelden_tot) and $groep->mag(AccessAction::Wijzigen)) {
 			ChangeLogModel::instance()->log($groep, 'aanmelden_tot', $groep->aanmelden_tot, getDateTime());
 			$groep->aanmelden_tot = getDateTime();
 			$this->model->update($groep);
 			$response[] = $groep;
 		}
-		$this->view = new GroepenBeheerData($response);
+		return new GroepenBeheerData($response);
 	}
 
-	public function voorbeeld(array $selection) {
-		if (empty($selection)) {
-			$this->exit_http(403);
-		}
+	public function voorbeeld($id) {
 		/** @var AbstractGroep $groep */
-		$groep = $this->model->retrieveByUUID($selection[0]);
+		$groep = $this->model->retrieveByUUID($id);
 		if (!$groep OR !$groep->mag(AccessAction::Bekijken)) {
-			$this->exit_http(403);
+			throw new CsrToegangException();
 		}
-		$this->view = new GroepPreviewForm($groep);
+		return new GroepPreviewForm($groep);
 	}
 
 	/**
-	 * @param AbstractGroep|null $groep
-	 * @throws CsrToegangException
+	 * @param Request $request
+	 * @param $id
+	 * @return GroepLogboekForm|GroepLogboekData
 	 */
-	public function logboek(AbstractGroep $groep = null) {
+	public function logboek(Request $request, $id) {
 		// data request
-		if ($groep) {
+		if ($request->getMethod() == 'POST') {
+			$groep = $this->model->get($id);
 			if (!$groep->mag(AccessAction::Bekijken)) {
-				$this->exit_http(403);
+				throw new CsrToegangException();
 			}
 			$data = ChangeLogModel::instance()->find('subject = ?', [$groep->getUUID()]);
-			$this->view = new GroepLogboekData($data);
+			return new GroepLogboekData($data);
 		} // popup request
 		else {
-			$selection = filter_input(INPUT_POST, 'DataTableSelection', FILTER_SANITIZE_STRING, FILTER_FORCE_ARRAY);
 			/** @var AbstractGroep $groep */
-			$groep = $this->model->retrieveByUUID($selection[0]);
+			$groep = $this->model->retrieveByUUID($id);
 			if (!$groep || !$groep->mag(AccessAction::Bekijken)) {
 				throw new CsrToegangException('Kan logboek niet vinden', 403);
 			}
-			$this->view = new GroepLogboekForm($groep);
+			return new GroepLogboekForm($groep);
 		}
 	}
 
-	public function leden(AbstractGroep $groep) {
+	public function leden(Request $request, $id) {
+		$groep = $this->model->get($id);
 		if (!$groep->mag(AccessAction::Bekijken)) {
-			$this->exit_http(403);
+			throw new CsrToegangException();
 		}
-		if ($this->getMethod() == 'POST') {
-			$this->view = new GroepLedenData($groep::getLedenModel()->getLedenVoorGroep($groep));
+		if ($request->getMethod() == 'POST') {
+			return new GroepLedenData($groep::getLedenModel()->getLedenVoorGroep($groep));
 		} else {
-			$this->view = new GroepLedenTable($groep::getLedenModel(), $groep);
+			return new GroepLedenTable($groep::getLedenModel(), $groep);
 		}
 	}
 
-	public function aanmelden2(AbstractGroep $groep, $uid) {
+	public function aanmelden2(Request $request, $id, $uid) {
```

Comment:
Dit is voor groepen V2 toch? Misschien even een comment boven zetten. Nu is het een beetje verwarrend.